spec/std/html_spec.cr

```diff
@@ -53,18 +53,33 @@ describe "HTML" do
       str.should eq("3 + 2 = 5")
     end
 
+    it "unescapes decimal encoded chars" do
+      str = HTML.unescape("3 &#00043; 2 &#00061 5")
+
+      str.should eq("3 + 2 = 5")
+    end
+
     it "unescapes &nbsp;" do
       str = HTML.unescape("nbsp&nbsp;space ")
 
       str.should eq("nbsp\u{0000A0}space ")
     end
 
-    it "unescapes Char::MAX_CODEPOINT" do
+    it "does not unescape Char::MAX_CODEPOINT" do
+      # Char::MAX_CODEPOINT is actually a noncharacter and is not replaced
       str = HTML.unescape("limit &#x10FFFF;")
-      str.should eq("limit 􏿿")
+      str.should eq("limit &#x10FFFF;")
 
       str = HTML.unescape("limit &#1114111;")
-      str.should eq("limit 􏿿")
+      str.should eq("limit &#1114111;")
+    end
+
+    it "unescapes characters above Char::MAX_CODEPOINT" do
+      str = HTML.unescape("limit &#x110000;")
```

Comment:
I don't get it, if 10FFFF is not a character, then why should something right after it be a character

Reply:
That's a typo. It doesn't decode `&#x110000;` but replaces it with replacement character `\uFFFD`.

Reply:
Question still stands. `&#x10FFFF;`->`&#x10FFFF;` but `&#x110000;` -> `\uFFFD` - why?

Reply:
`&#x10FFFF;` is a valid codepoint, yet not supported as a character reference because it's a unicode noncharacter. Therefore it is not decoded.
`&#x110000;` is an invalid codepoint and therefore replaced by replacement character.

See [*Numeric character reference end state*](https://html.spec.whatwg.org/multipage/parsing.html#numeric-character-reference-end-state) for details.